activation/handler.go

```diff
@@ -562,10 +563,14 @@ func (h *Handler) handleAtx(
 	msg []byte,
 ) (*types.MalfeasanceProof, error) {
 	receivedTime := time.Now()
-	var atx types.ActivationTx
-	if err := codec.Decode(msg, &atx); err != nil {
+
+	var atxOnWire wire.ActivationTxV1
+	if err := codec.Decode(msg, &atxOnWire); err != nil {
 		return nil, fmt.Errorf("%w: %w", errMalformedData, err)
 	}
+
+	atx := types.ActivationTxFromWireV1(&atxOnWire)
```

Comment:
I also need it in the SQL code to deserialize the blob. Hopefully, we will remove decoding the blob soon and then it can be moved to the ATX handler code.

Reply:
I think the `ActivationTxFromWireV1` and `ActivationTxFromWireV1` generally should not be needed. We should never convert from Domain to wire type (the builder works on the wire type directly) and the conversion from wire to domain is only needed in the handler in one place - after successful validation and before storing the ATX in the DB.

All tests should either only use the wire type (e.g. Builder and Handler test) or only use the domain type (pretty much everything else that needs ATXs)